tinyiiod.c

```diff
@@ -62,160 +74,219 @@ int tinyiiod_read_command(struct tinyiiod *iiod)
 {
-	iiod->ops->read(buf, len);
+	return iiod->ops->read(&iiod->instance_id, buf, len);
 }
 
-ssize_t tinyiiod_read_line(struct tinyiiod *iiod, char *buf, size_t len)
+/***************************************************************************//**
+ * @brief tinyiiod_read_line
+*******************************************************************************/
+int32_t tinyiiod_read_line(struct tinyiiod *iiod, char *buf)
 {
-	unsigned int i;
-	bool found = false;
-
-	for (i = 0; i < len - 1; i++) {
-		buf[i] = tinyiiod_read_char(iiod);
-
-		if (buf[i] != '\n' && buf[i] != '\r')
-			found = true;
-		else if (found)
-			break;
-	}
+	return iiod->ops->read_line(&iiod->instance_id, buf);
+}
 
-	if (!found || i == len - 1) {
-		/* No \n found -> garbage data */
-		return -EIO;
-	}
+/***************************************************************************//**
+ * @brief tinyiiod_read_nonblocking
+*******************************************************************************/
+int32_t tinyiiod_read_nonblocking(struct tinyiiod *iiod, char *buf, size_t len)
+{
+	return iiod->ops->read_nonbloking(&iiod->instance_id, buf, len);
+}
 
-	buf[i] = '\0';
-	return (ssize_t) i;
+/***************************************************************************//**
+ * @brief tinyiiod_read_wait
+*******************************************************************************/
+int32_t tinyiiod_read_wait(struct tinyiiod *iiod, size_t len)
+{
+	return iiod->ops->read_wait(&iiod->instance_id, len);
 }
 
+/***************************************************************************//**
+ * @brief tinyiiod_write_char
+*******************************************************************************/
 void tinyiiod_write_char(struct tinyiiod *iiod, char c)
 {
-	iiod->ops->write(&c, 1);
+	iiod->ops->write(iiod->instance_id, &c, 1);
 }
 
+/***************************************************************************//**
+ * @brief tinyiiod_write
+*******************************************************************************/
 void tinyiiod_write(struct tinyiiod *iiod, const char *data, size_t len)
 {
-	iiod->ops->write(data, len);
+	iiod->ops->write(iiod->instance_id, data, len);
 }
 
+/***************************************************************************//**
+ * @brief tinyiiod_write_string
+*******************************************************************************/
 void tinyiiod_write_string(struct tinyiiod *iiod, const char *str)
 {
 	tinyiiod_write(iiod, str, strlen(str));
 }
 
-void tinyiiod_write_value(struct tinyiiod *iiod, int value)
+/***************************************************************************//**
+ * @brief tinyiiod_write_value
+*******************************************************************************/
+void tinyiiod_write_value(struct tinyiiod *iiod, int32_t value)
 {
 	char buf[16];
 
-	snprintf(buf, sizeof(buf), "%i\n", value);
+	snprintf(buf, sizeof(buf), "%"PRIi32"\n", value);
```

Comment:
the `PRIi32` macro may need to be put into a `compat.h` file and checked whether `#ifndef PRIi32`
i am not sure how standard this is